### ISSUE TITLE
Generated error encoder doesn't handle design errors correctly

### ISSUE THREAD

**Issue by khaledez** [goadesign/goa#3156]
**Affected version: `3.9.0`**

I have this design:
```go
var WaitListSignUpPayload = Type("WaitListSignUpPayload", func() {
	Attribute("email", String, func() {
		Description("Customer e-mail")
		Format(FormatEmail)
		MaxLength(128)
	})

	Required("email")
})

var _ = Service("wait-list", func() {
	Description("Early adopters wait list")

	Error("BadRequest")

	HTTP(func() {
		Response("BadRequest", StatusBadRequest)
	})

	Method("sign-up", func() {
		Description("Sign-up for the waitlist")
		Payload(func() {
			Extend(WaitListSignUpPayload)
		})
		Error("Unprocessable")
		HTTP(func() {
			POST("/waitlist/sign-up")
			Response("Unprocessable", StatusUnprocessableEntity)
			Response(StatusCreated)
		})
	})

})
```

when I generate the endpoints and types and make a request that expects to return a bad request, it return "internal error" instead.

After investigating, I found that the generated http `EncodeSignUpError` (inside `gen/http/wait_list/server/encode_decode.go`) is causing the issue.

When I removed the first 4 lines of the function, it worked as expected. These are the lines:
```
	var deprecatedErrorNamer ErrorNamer
	if errors.As(v, &deprecatedErrorNamer) {
		v = adaptErrorNamer{deprecatedErrorNamer}
	}
```

These lines were introduced in #3147 by @ernesto-jimenez

**Comment by khaledez** [comment]
if you want to test it with code, download the following example project, unzip it, then cd into it and:
```
go generate ./...
```

[goa390-error.zip](https://github.com/goadesign/goa/files/9697496/goa390-error.zip)

**Comment by ernesto-jimenez** [comment]
Is this addressed by #3155?

**Comment by khaledez** [comment]
> Is this addressed by #3155?

@ernesto-jimenez no. It doesn't solve the issue!

**Comment by raphael** [comment]
@ernesto-jimenez I think we should be able to remove the `adapter` stuff. The new code generated by Goa won't make use of `ErrorName()` by definition and the error data structures will expose `GoaErrorName()`. So, it should all be consistent within a single `gen` folder.

**Comment by raphael** [comment]
Put together https://github.com/goadesign/goa/pull/3157 to remove the adapter.

**Comment by raphael** [comment]
Merged #3157 - will release `3.9.1` with the fix.

**Comment by khaledez** [comment]
Thanks @raphael The issue is solved
